translate/providers.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+# encoding: utf-8
+# ----------------------------------------------------------------------------
+# "THE BEER-WARE LICENSE" (Revision 42):
+# <terry.yinzhe@gmail.com> wrote this file. As long as you retain this notice you
+# can do whatever you want with this stuff. If we meet some day, and you think
+# this stuff is worth it, you can buy me a beer in return to Terry Yin.
+from abc import ABCMeta, abstractmethod
+
+import requests
+
+
+class BaseProvider:
+    __metaclass__ = ABCMeta
+
+    base_url = ''
+
+    def __init__(self, to_lang, headers, from_lang='en', **kwargs):
+        self.from_lang = from_lang
+        self.to_lang = to_lang
+        self.headers = headers
+
+    @abstractmethod
+    def _make_request(self, params):
+        return NotImplemented('Please Implement this method')
+
+    @abstractmethod
+    def get_translation(self, params):
+        return NotImplemented('Please Implement this method')
+
+
+class MyMemoryProvider(BaseProvider):
+    base_url = 'http://mymemory.translated.net/api/get'
+
+    def __init__(self, **kwargs):
+        try:
+            super().__init__(**kwargs)
```

Comment:
This is because python version differences. here is > 3.x versions and on line 39, 2x versions